src/testing/core/__tests__/createTestSchema.test.tsx

```diff
@@ -1033,7 +1024,104 @@ describe("schema proxy", () => {
         },
       });
     }
+  });
 
-    unmount();
+  it("createSchemaFetch respects min and max delay", async () => {
+    const Profiler = createDefaultProfiler<ViewerQueryData>();
+
+    const minDelay = 1500;
+    const maxDelay = 2000;
+
+    using _fetch = createSchemaFetch(schema, {
+      delay: { min: minDelay, max: maxDelay },
+    }).mockGlobal();
+
+    const client = new ApolloClient({
+      cache: new InMemoryCache(),
+      uri,
+    });
+
+    const query: TypedDocumentNode<ViewerQueryData> = gql`
+      query {
+        viewer {
+          id
+          name
+          age
+          book {
+            id
+            title
+            publishedAt
+          }
+        }
+      }
+    `;
+
+    const Fallback = () => {
+      useTrackRenders();
+      return <div>Loading...</div>;
+    };
+
+    const App = () => {
+      return (
+        <React.Suspense fallback={<Fallback />}>
+          <Child />
+        </React.Suspense>
+      );
+    };
+
+    const Child = () => {
+      const result = useSuspenseQuery(query);
+
+      useTrackRenders();
+
+      Profiler.mergeSnapshot({
+        result,
+      } as Partial<{}>);
+
+      return <div>Hello</div>;
+    };
+
+    renderWithClient(<App />, {
+      client,
+      wrapper: Profiler,
+    });
+
+    // initial suspended render
+    await Profiler.takeRender();
+
+    await expect(Profiler).not.toRerender({ timeout: minDelay - 100 });
+
+    {
+      const { snapshot } = await Profiler.takeRender({
+        timeout: maxDelay + 100,
+      });
```

Comment:
```suggestion
    {
      const { snapshot } = await Profiler.takeRender({
        timeout: maxDelay - minDelay + 110,
      });
```
This timeout starts after `minDelay - 100` has passed. Adding another 10ms to prevent flakyness.

Reply:
I see what you mean! Updated and added a comment here https://github.com/apollographql/apollo-client/pull/11774/commits/06b43892d983f222e7c7e24ea3367d0e3dab68f8, thanks :)